pysat/instruments/sw_f107.py

```diff
@@ -459,46 +474,54 @@ def download(date_array, tag, sat_id, data_path, user=None, password=None):
                     checkfile = os.path.split(outfile)[-1]
                     has_file = local_files == checkfile
                     if np.any(has_file):
-                        if has_file[has_file].index[-1] < today or got_today:
-                            rewritten = True
-                            break
+                        if has_file[has_file].index[-1] < vend[iname]:
+                            # This file will be updated again, but only attempt
+                            # to do so if enough time has passed from the
+                            # last time it was downloaded
+                            yesterday = today - pds.DateOffset(days=1)
+                            if has_file[has_file].index[-1] < yesterday:
+                                rewritten = True
+
+                # Attempt to download if the file does not exist or if the
+                # file has been updated
+                if rewritten or not downloaded:
+                    try:
+                        sys.stdout.flush()
+                        ftp.retrbinary('RETR ' + fname,
+                                       open(saved_fname, 'wb').write)
+                        downloaded = True
+                        print('Downloaded file for ' + date.strftime('%x'))
+
+                    except ftplib.error_perm as exception:
+                        # Could not fetch, so cannot rewrite
+                        rewritten = False
+
+                        # Test for an error
+                        if str(exception.args[0]).split(" ", 1)[0] != '550':
+                            raise RuntimeError(exception)
                         else:
-                            got_today = True
-
-                try:
-                    print('Downloading file for ' + date.strftime('%x'))
-                    sys.stdout.flush()
-                    ftp.retrbinary('RETR ' + fname,
-                                   open(saved_fname, 'wb').write)
-                    downloaded = True
+                            # file isn't actually there, try the next name
+                            os.remove(saved_fname)
```

Comment:
Should this and the next line precede the raise error statement?  Not clear these will be run.

Reply:
No, because this loop looks for two possible files and it's ok if one isn't there.

Reply:
OK.  The diff file threw me off.  Actual file looks ok